src/target/riscv/riscv.h

```diff
@@ -370,8 +373,10 @@ int riscv_current_hartid(const struct target *target);
  * consecutive and start with mhartid=0. */
 unsigned int riscv_count_harts(struct target *target);
 
-/** Set register, updating the cache. */
+/** Set register, if possible, only updating the cache.*/
 int riscv_set_register(struct target *target, enum gdb_regno i, riscv_reg_t v);
+/** Write register to target, updating the cache. */
+int riscv_write_register(struct target *target, enum gdb_regno i, riscv_reg_t v);
```

Comment:
I tried to make the comments even more clear:

```suggestion
/** Set the register value. For cacheable registers, only the cache is updated (write-back mode). */
int riscv_set_register(struct target *target, enum gdb_regno i, riscv_reg_t v);
/** Set the register value and immediately write it to the target (write-through mode). 
    Update the cache (for cacheable registers only). */
int riscv_write_register(struct target *target, enum gdb_regno i, riscv_reg_t v);
```

Reply:
Addressed.